test/integration/appfactory/appfactory_test.go

```diff
@@ -15,23 +15,135 @@
 package appfactory
 
 import (
+	"fmt"
 	"testing"
 	"time"
 
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/gcloud"
 	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/tft"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/utils"
+	"github.com/gruntwork-io/terratest/modules/terraform"
 	"github.com/stretchr/testify/assert"
+
 	"github.com/terraform-google-modules/enterprise-application/test/integration/testutils"
 )
 
 func TestAppfactory(t *testing.T) {
-	bootstrap := tft.NewTFBlueprintTest(t,
+	appFactory := tft.NewTFBlueprintTest(t,
 		tft.WithTFDir("../../../3-appfactory/apps"),
 		tft.WithRetryableTerraformErrors(testutils.RetryableTransientErrors, 3, 2*time.Minute),
 	)
 
-	bootstrap.DefineVerify(func(assert *assert.Assertions) {
-		bootstrap.DefaultVerify(assert)
+	appFactory.DefineVerify(func(assert *assert.Assertions) {
+		appFactory.DefaultVerify(assert)
+
+		// check admin project
+		adminProjectID := appFactory.GetStringOutput("app_admin_project_id")
+		adminProjectApis := []string{
+			"iam.googleapis.com",
+			"cloudresourcemanager.googleapis.com",
+			"cloudbuild.googleapis.com",
+			"secretmanager.googleapis.com",
+			"serviceusage.googleapis.com",
+			"cloudbilling.googleapis.com",
+			"cloudfunctions.googleapis.com",
+			"apikeys.googleapis.com",
+			"sourcerepo.googleapis.com",
+		}
+
+		prj := gcloud.Runf(t, "projects describe %s", adminProjectID)
+		assert.Equal("ACTIVE", prj.Get("lifecycleState").String(), fmt.Sprintf("project %s should be ACTIVE", adminProjectID))
+
+		enabledAPIS := gcloud.Runf(t, "services list --project %s", adminProjectID).Array()
+		listApis := testutils.GetResultFieldStrSlice(enabledAPIS, "config.name")
+		assert.Subset(listApis, adminProjectApis, "APIs should have been enabled")
+
+		// check app infra repo
+		repositoryName := appFactory.GetStringOutput("app_infra_repository_name")
+		repoURL := fmt.Sprintf("https://source.developers.google.com/p/%s/r/%s", adminProjectID, repositoryName)
+		repoOP := gcloud.Runf(t, "source repos describe %s --project %s", repositoryName, adminProjectID)
+		assert.Equal(repoURL, repoOP.Get("url").String(), "source repo %s should have url %s", repositoryName, repoURL)
+
+		// check workspace SA access to repo
+		repoSa := fmt.Sprintf("serviceAccount:tf-cb-%s@%s.iam.gserviceaccount.com", repositoryName, adminProjectID)
+		repoIamOpts := gcloud.WithCommonArgs([]string{"--flatten", "bindings", "--filter", "bindings.role:roles/viewer", "--format", "json"})
+		repoIamPolicyOp := gcloud.Run(t, fmt.Sprintf("source repos get-iam-policy %s --project %s", repositoryName, adminProjectID), repoIamOpts).Array()[0]
```

Comment:
Will `gcloud source repos get-iam-policy` always return a single element? 

I ask because of `[0]`. Maybe cloud be nice to have a `--sort-by` or `--filter` to make sure we are indexing the right record.

Reply:
it will return only one element because it already has a filter
`--filter "bindings.role:roles/viewer"`

Reply:
Oh, I didn't notice that before. Thanks for pointing!